application/common/models/User.php

```diff
@@ -0,0 +1,190 @@
+
+    public $password;
+
+    public function scenarios(): array
+    {
+        $scenarios = parent::scenarios();
+
+        $scenarios[self::SCENARIO_DEFAULT] = [
+            'employee_id',
+            'first_name',
+            'last_name',
+            'display_name',
+            'username',
+            'email',
+            'active',
+            'locked',
+        ];
+
+        $scenarios[self::SCENARIO_UPDATE_PASSWORD] = ['password'];
+
+        $scenarios[self::SCENARIO_AUTHENTICATE] = ['username', 'password'];
+
+        return $scenarios;
+    }
+
+    public function rules(): array
+    {
+        return ArrayHelper::merge([
+            [
+                'active', 'default', 'value' => 'yes',
+            ],
+            [
+                'locked', 'default', 'value' => 'no',
+            ],
+            [
+                ['active', 'locked'], 'in', 'range' => ['yes', 'no'],
+            ],
+            [
+                'email', 'email',
+            ],
+            [
+                'password', 'required',
+                'on' => [self::SCENARIO_UPDATE_PASSWORD, self::SCENARIO_AUTHENTICATE],
+            ],
+            [
+                'password', 'string',
+            ],
+            [
+                'password',
+                $this->validatePassword(),
+                'on' => self::SCENARIO_AUTHENTICATE,
+            ],
+            [
+                ['last_synced_utc', 'last_changed_utc'],
+                'default', 'value' => Utils::now(),
+            ],
+            [
+                ['employee_id', 'first_name', 'last_name', 'display_name', 'username', 'email'],
+                'string',
+            ],
+        ], parent::rules());
+    }
+
+    private function validatePassword(): \Closure
+    {
+        return function ($attributeName) {
+            if (! password_verify($this->$attributeName, $this->password_hash)) {
+                $this->addError($attributeName, 'Incorrect password.');
+            }
+        };
+    }
+
+    public function behaviors(): array
+    {
+        return [
+            'changeTracker' => [
+                'class' => AttributeBehavior::className(),
+                'attributes' => [
+                    ActiveRecord::EVENT_BEFORE_INSERT => 'last_changed_utc',
+                    ActiveRecord::EVENT_BEFORE_UPDATE => 'last_changed_utc',
+                ],
+                'value' => Utils::now(),
+                'skipUpdateOnClean' => true, // only update the column if the model is dirty
+            ],
+            'updateTracker' => [
+                'class' => AttributeBehavior::className(),
+                'attributes' => [
+                    ActiveRecord::EVENT_BEFORE_INSERT => 'last_synced_utc',
+                    ActiveRecord::EVENT_BEFORE_UPDATE => 'last_synced_utc',
+                ],
+                'value' => Utils::now(),
+                'skipUpdateOnClean' => false, // always update the column regardless of dirtiness
```

Comment:
Are there scenarios where the model will change (such as a new password pushed from the password-reset tool) but where we _don't_ want `last_synced_utc` to change? I think `last_synced_utc` is just for when this record has been synchronized with the ID Store, so changing the password shouldn't update `last_synced_utc` (as far as I understand things).